shim/stub.go

```diff
@@ -166,6 +166,16 @@ func (s *ChaincodeStub) GetState(key string) ([]byte, error) {
 	return s.handler.handleGetState(collection, key, s.ChannelID, s.TxID)
 }
 
+// GetStateMultipleKeys documentation can be found in interfaces.go
+func (s *ChaincodeStub) GetStateMultipleKeys(keys []string) ([][]byte, error) {
+	// Access public data by setting the collection to empty string
+	collection := ""
+	if len(keys) == 0 {
+		return nil, fmt.Errorf("keys must not be an empty slice")
+	}
```

Comment:
I am not sure this needs to be an error condition. If they ask for no keys, we could return them no results rather than requiring error handling.

```suggestion
	if len(keys) == 0 {
		return nil, nil
	}
```

Reply:
You can do that. But it seems to me that if we are called without passing any key, then the caller has an error somewhere. How do we signal him?

Reply:
I'm not sure it's our job to try to flag potential errors in application logic. Maybe it's not even an error. Maybe the transaction function allows a set of key fragments to be passed in, which the transaction function then assembles into a set of key names and gets from the ledger. If no key fragments are passed in, an empty set of values is validly returned.

If we stretch the error handling argument further, perhaps it should also be an error to have a single key since they should have used `GetState` for a single key? Should the check be `if len(keys) < 2`? It is just flagging a potential error in application logic like the empty slice case, right?

Looking at the handler code, an empty slice of keys will still return an empty slice, so perhaps there doesn't need to be any special case code for empty keys here at all.

Reply:
done

Reply:
> I'm not sure it's our job to try to flag potential errors in application logic. Maybe it's not even an error. Maybe the transaction function allows a set of key fragments to be passed in, which the transaction function then assembles into a set of key names and gets from the ledger. If no key fragments are passed in, an empty set of values is validly returned.

Yet, we return, for instance, an error on missing collection name while calling private data APIs.

Reply:
That is not a _potential_ error in application logic. That is a hard error in the API usage. You cannot access a private data collection without giving the name of the collection you want to access.